docs/src/main/mdoc/datatypes/statet.md

```diff
@@ -109,52 +126,56 @@ val gameState2 = GameState.attemptGuess(gameState1, "bats")
 println(GameState.showWordHidden(gameState2.getOrElse(???)))
 ```
 
-Let's focus on the state machine game steps.
+Let's focus on the state machine game steps:
 
 ```scala mdoc:silent
 object StateMachine {
 
-  def step1(): Eval[GameState] = for {
-    _ <- Eval.later(println("Insert the target word: "))
-    word <- Eval.later(readLine())
+  import EffectType._
+
+  def step1(): EffectType[GameState] = for {
+    _ <- performEffect[Unit](println("Insert the target word: "))
+    word <- performEffect[String](readLine())
```

Comment:
Sorry, I think there was a miscommunication :) since we don't have `IO`, there's really no way we can use `readLine()` or `printLine()` or any side-effects at all in this example unfortunately.

One way forward is to use effects that we can model with datatypes available in cats. For example, `Either` can be used to raise errors and short circuit, or `Writer` can be used for logging (this would be the closest to what you currently have).

Reply:
Ah ok, 
Now everything is clearer. 
Then, I'll proceed in this way:
- Refactor the code for the hangman example using `IO`, but not embedding the code into the docs. Just put an external link to the gist. Otherwise, I can just add the code, but ask `mdoc` to not compile/run it, even if it might expose it to go out of sync.
- Add another example with `Either` and some validation process that require a State as well.

Let me know your thoughts about this plan, if you like it, then I'll proceed 😉
Thanks for the feedback 🙏

Reply:
👍 to the table reservation example, I like it! :)

I think we can keep your current example (refactored to use `IO`) as a second, "advanced" example like you suggest. Instead of a gist I wonder if we could put it in a scastie.